build regression in 2.51.4_rc3: IN_EXCL_UNLINK

2e12fc4f41b39ca9a5e501d7df5c2c33a75d3b95 builds

eae947ee7a6076592ec711e4c090438ab6ead211 fails

````
....
[   70s] /usr/src/packages/BUILD/unison-2.51.4_rc3/fsmonitor/linux/inotify_stubs.c:36: error: 'IN_EXCL_UNLINK' undeclared here (not in a function)
[   70s] make: *** [fsmonitor/linux/inotify_stubs.o] Error 2
[   70s] error: Bad exit status from /var/tmp/rpm-tmp.CrByRI (%build)
````

See old dists at [unison:next](https://build.opensuse.org/package/show/Archiving:unison:next/unison).

`IN_EXCL_UNLINK` was added by torvalds/linux@8c1934c8d70b22ca8333b216aec6c7d09fdbd6a6 in v2.6.36-rc1. At some point it was also added to glibc.

Maybe unison wants to support old userspace which lacks this knob?

--- a/src/fsmonitor/linux/inotify_stubs.c
+++ b/src/fsmonitor/linux/inotify_stubs.c
@@ -28,6 +28,10 @@
 #include <caml/signals.h>
 #include <caml/callback.h>
 
+#ifndef IN_EXCL_UNLINK
+#define IN_EXCL_UNLINK 0  /* If not supported, just ignore */
+#endif
+
 static int inotify_flag_table[] = {
         IN_ACCESS, IN_ATTRIB, IN_CLOSE_WRITE, IN_CLOSE_NOWRITE,
         IN_CREATE, IN_DELETE, IN_DELETE_SELF, IN_MODIFY,

